src/app/service-offering/custom-service-offering/custom-service-offering.component.ts

```diff
@@ -35,11 +57,10 @@ export class CustomServiceOfferingComponent {
   }
 
   private createForm() {
-    // input text=number provide all other validation for current restrictions
     this.hardwareForm = new FormGroup({
-      cpuNumber: new FormControl(this.offering.cpunumber, Validators.required),
-      cpuSpeed: new FormControl(this.offering.cpuspeed, Validators.required),
-      memory: new FormControl(this.offering.memory, Validators.required),
+      cpuNumber: new FormControl(this.offering.cpunumber, [Validators.required, LimitValidator(this.maxCpu)]),
+      cpuSpeed: new FormControl(this.offering.cpuspeed, [Validators.required]),
+      memory: new FormControl(this.offering.memory, [Validators.required, LimitValidator(this.maxMemory)]),
```

Comment:
The code does not work without this check? 
The comment `// input text=number provide all other validation for current restrictions` describe that all validation provided by input itself. I think it should be placed back. You add only validation for max value, but input provide both max and min. See #1082

Reply:
To discuss